engine/access/rpc/backend/backend_transactions_test.go

```diff
@@ -0,0 +1,326 @@
+		}
+
+		suite.historicalAccessClient.
+			On("GetTransactionResult", mock.Anything, mock.Anything).
+			Return(&transactionResultResponse, nil).Once()
+
+		backend := New(Params{
+			State:                 suite.state,
+			CollectionRPC:         suite.colClient,
+			HistoricalAccessNodes: []access.AccessAPIClient{suite.historicalAccessClient},
+			Blocks:                suite.blocks,
+			Transactions:          suite.transactions,
+			ExecutionReceipts:     suite.receipts,
+			ChainID:               suite.chainID,
+			AccessMetrics:         metrics.NewNoopCollector(),
+			MaxHeightRange:        DefaultMaxHeightRange,
+			Log:                   suite.log,
+			SnapshotHistoryLimit:  DefaultSnapshotHistoryLimit,
+			Communicator:          suite.communicator,
+			TxResultCacheSize:     10,
+		})
+
+		coll := flow.CollectionFromTransactions([]*flow.Transaction{tx})
+
+		resp, err := backend.GetTransactionResult(context.Background(), tx.ID(), block.ID(), coll.ID())
+		suite.Require().NoError(err)
+		suite.Require().Equal(flow.TransactionStatusExecuted, resp.Status)
+		suite.Require().Equal(uint(flow.TransactionStatusExecuted), resp.StatusCode)
+
+		resp2, err := backend.GetTransactionResult(context.Background(), tx.ID(), block.ID(), coll.ID())
+		suite.Require().NoError(err)
+		suite.Require().Equal(flow.TransactionStatusExecuted, resp2.Status)
+		suite.Require().Equal(uint(flow.TransactionStatusExecuted), resp2.StatusCode)
+
+		suite.historicalAccessClient.AssertExpectations(suite.T())
+	})
+}
+
+func (suite *Suite) TestGetTransactionResultCacheNonExistent() {
+	suite.WithGetTransactionCachingTestSetup(func(block *flow.Block, tx *flow.Transaction) {
+
+		suite.historicalAccessClient.
+			On("GetTransactionResult", mock.Anything, mock.Anything).
+			Return(nil, status.Errorf(codes.NotFound, "no known transaction with ID %s", tx.ID())).Once()
+
+		backend := New(Params{
+			State:                 suite.state,
+			CollectionRPC:         suite.colClient,
+			HistoricalAccessNodes: []access.AccessAPIClient{suite.historicalAccessClient},
+			Blocks:                suite.blocks,
+			Transactions:          suite.transactions,
+			ExecutionReceipts:     suite.receipts,
+			ChainID:               suite.chainID,
+			AccessMetrics:         metrics.NewNoopCollector(),
+			MaxHeightRange:        DefaultMaxHeightRange,
+			Log:                   suite.log,
+			SnapshotHistoryLimit:  DefaultSnapshotHistoryLimit,
+			Communicator:          suite.communicator,
+			TxResultCacheSize:     10,
+		})
+
+		coll := flow.CollectionFromTransactions([]*flow.Transaction{tx})
+
+		resp, err := backend.GetTransactionResult(context.Background(), tx.ID(), block.ID(), coll.ID())
+		suite.Require().NoError(err)
+		suite.Require().Equal(flow.TransactionStatusUnknown, resp.Status)
+		suite.Require().Equal(uint(flow.TransactionStatusUnknown), resp.StatusCode)
+
+		// ensure the unknown transaction is cached when not found anywhere
```

Comment:
```suggestion
		// ensure the unknown transaction is not cached when not found anywhere
```

Reply:
well we actually DO cache transactions that are not found as per https://github.com/onflow/flow-go/blob/a59b7630db01d5c32cb5a53390f95963094228af/engine/access/rpc/backend/backend_transactions.go#L259